### PR TITLE
Adds Table of Content to the awesome list

### PR DESCRIPTION
**Pull request by ajeetraina**
This PR adds Table of Contents to the README file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,12 +10,26 @@ This is a curated list of Testcontainers related resources. It is not an officia
...
+12. [Testcontainers and Spring](#testcontainers-and-spring)
+13. [Integrations](#integrations)
+14. [Blogs](#blogs)
+15. [Featured Videos](#featured-videos)
+16. [Community](#community)
 
-- Join 9900+ DevOps Engineers today via [Community Slack](https://launchpass.com/collabnix)
-- Join our [Discord Server](https://discord.gg/QEkCXAXYSe)
-- Fork, Contribute & Share via [Testcontainers GITHUB Repository](https://github.com/collabnix/awesome-testcontainers)
-- Click and Follow us over Twitter [![Twitter URL](https://img.shields.io/twitter/url/https/twitter.com/fold_left.svg?style=social&label=Follow%20%40collabnix)](https://twitter.com/collabnix)
```
Comment (mdelapenya) [inline README.md:18]: Do we want to remove these social links?

Comment (ajeetraina) [reply README.md:18]: Yes, we can remove these sections. The intention is not to target a specific community but to open it to the global community. We will get testcontainers Slack channel added later point of time. I will get it added under Community section.

File: README.md [hunk 2]
```diff
@@ -10,12 +10,26 @@ This is a curated list of Testcontainers related resources. It is not an officia
...
+6. [Testcontainers Desktop](#testcontainers-desktop)
+7. [Testcontainers and C/C++/Swift](#testcontainers-and-ccswift)
+8. [Testcontainers and Kubernetes](#testcontainers-and-kubernetes)
+9. [Testcontainers and .NET](#testcontainers-and-net)
+10. [Testcontainers and Go](#testcontainers-and-go)
+11. [Testcontainers and PHP](#testcontainers-and-php)
+12. [Testcontainers and Spring](#testcontainers-and-spring)
+13. [Integrations](#integrations)
+14. [Blogs](#blogs)
+15. [Featured Videos](#featured-videos)
```
Comment (mdelapenya) [inline README.md:30]: In #32 we added the Developer Stories section
```suggestion
14. [Blogs](#blogs)
15. [Developer Stories](#developer-stories)
16. [Featured Videos](#featured-videos)
```

**Review by mdelapenya — APPROVED** [review]
LGTM, thanks!

**Pull request merged**